### DISCUSSION TITLE
Syntax Highlighting for <style> inside .html files

### DISCUSSION THREAD

**Discussion by octoxan** [LazyVim/LazyVim#930] [Q&A] [answered]
Following along with the LazyVim YouTube series and they start off by opening a TypeScript project in nvim. I don’t have one of those so I just opened a HubSpot project and noticed that while most of the .html file is correctly highlighted, everything inside the <style> blocks are not.

I opened mason and searched for the css lsp and installed it. Then I tried creating a plain old html document (not HubSpot weirdness lol) with a head, body, and a <style> block in the head and get the same thing.

I can manually set the file type to CSS and then it works but disables all the HTML syntax highlighting.

Is there a proper way to have both? Ideally I’ll just never touch another HubSpot project and can keep my CSS in their own files, but while learning I figure this is good to know if it’s something I can configure.

**Comment by dpetka2001** [comment, marked as answer]
This might be treesitter highlighting. Do you have css parser for Treesitter installed? `:TSInstall css` should do it. You might also want to include it in your Treesitter spec in `ensure_installed`, so that it gets automatically installed if you move your configuration to another computer, so you won't have to manually install it again.
